spec/models/invoice_spec.rb

```diff
@@ -1765,4 +1765,45 @@
       end
     end
   end
+
+  describe "should_apply_provider_tax?" do
+    let(:invoice) { create(:invoice, :subscription, subscriptions: [subscription]) }
+    let(:plan) { create(:plan) }
+    let(:subscription) { create(:subscription, plan:) }
+
+    let(:charge) { create(:standard_charge, plan:) }
+    let(:charge_fee) { create(:charge_fee, charge:, invoice:, amount: 100) }
+
+    before { charge_fee }
+
+    context "when there are fees with non zero amount" do
+      it { expect(invoice).to be_should_apply_provider_tax }
+    end
+
+    context "when there are no fees" do
+      let(:charge_fee) { nil }
+
+      it { expect(invoice).not_to be_should_apply_provider_tax }
```

Comment:
A matter of taste but maybe you'll like it also:
Add `subject { invoice.should_apply_provider_tax? }` right after `describe`.

Then change it to such format.
```suggestion
      it { expect(subject).to be false }
```

or `it { expect(subject).to be true }`.

If at any moment we'll decide to rename this method we'd only need to change it once in `subject` instead of all tests.